### PR TITLE
Update room and chat api. Closes #63 and Closes #64

### LINKED ISSUE #63: Update Chat API call backend

### LINKED ISSUE #64: Update Room Selection API Calls backend

### PR DESCRIPTION
**Pull request by GijsWeterings**

### CODE REVIEW

File: src/main/java/me/moodcat/api/RoomAPI.java [hunk 1]
```diff
@@ -29,23 +37,63 @@
...
+        } else if (limit == 1) {
+            return allRooms.subList(0, 1);
+        } else if (limit > 1) {
+            return allRooms.subList(0, limit);
+        }
+
+        return allRooms;
     }
 
     @GET
```
Comment (newnottakenname) [inline src/main/java/me/moodcat/api/RoomAPI.java]: Javadoc?

File: src/main/java/me/moodcat/mood/Mood.java [hunk 2]
```diff
@@ -8,19 +10,20 @@
...
+    EXCITING(new VAVector(0.0, 0.0), "Exiting"),
+    HAPPY(new VAVector(1.0, 1.0), "Happy"),
+    NERVOUS(new VAVector(0.0, 0.0), "Nervous"),
+    PLEASING(new VAVector(0.0, 0.0), "Pleasing"),
+    PEACEFUL(new VAVector(0.0, 0.0), "Peaceful"),
+    RELAXED(new VAVector(0.0, 0.0), "Relaxed"),
+    SAD(new VAVector(0.0, 0.0), "Sad"),
+    SLEEPY(new VAVector(0.0, 0.0), "Sleepy");
 
     // CHECKSTYLE:ON
```
Comment (newnottakenname) [inline src/main/java/me/moodcat/mood/Mood.java]: CHEATER

File: src/main/java/me/moodcat/database/MockData.java [hunk 3]
```diff
@@ -56,11 +56,22 @@ public void insertMockData() {
         songDAO.persist(song);
 
         final Room room = new Room();
-        room.setCurrentSong(song);
+        room.setSong(song);
         room.setPosition(4);
-        room.setCurrentTime(42);
-        room.setRoomName("Silver Ass Wankers");
+        room.setTime(42);
+        room.setName("Silver Ass Wankers");
```
Comment (newnottakenname) [inline src/main/java/me/moodcat/database/MockData.java]: MAAAAAAAAAAAAAAAaaaaaaaaaaaaaaaaaaAAAAAAAAAAAAAAAAAAAAAaaaaaaaaaaaaaaaaaAaaaaaaaaaaaaaAAAAAAAAAAAAAAAAAAAAAaAAAAAAAAAAAAAAaaaaaaaaaaaaaaaaaaagic!

-Job

**Comment by Azkle** [comment]
The code seems ok.
I already worked with this branch and didn't found anything unusual.

Good to merge, for me.

File: src/main/java/me/moodcat/database/MockData.java [hunk 4]
```diff
@@ -56,11 +56,22 @@ public void insertMockData() {
...
         roomDAO.persist(room);
+
+        final Room room2 = new Room();
+        room2.setSong(song);
+        room2.setPosition(1);
+        room2.setTime(25);
+        room2.setName("Negative");
+        room2.setArousal(0.5);
+        room2.setValence(0.5);
+        roomDAO.persist(room2);
```
Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/database/MockData.java]: Mock data is removed in another branch..

**Comment by JWGmeligMeyling** [comment]
Looks good can be merged! :+1:

File: src/main/java/me/moodcat/mood/Mood.java [hunk 5]
```diff
@@ -8,19 +10,20 @@
...
-    CALM(0.0, 0.0),
-    EXCITING(0.0, 0.0),
-    HAPPY(0.0, 0.0),
-    NERVOUS(0.0, 0.0),
-    PLEASING(0.0, 0.0),
-    PEACEFUL(0.0, 0.0),
-    RELAXED(0.0, 0.0),
-    SAD(0.0, 0.0),
-    SLEEPY(0.0, 0.0);
+    ANGRY(new VAVector(-1.0, -1.0), "Angry"),
```
Comment (JeremybellEU) [inline src/main/java/me/moodcat/mood/Mood.java]: Maybe it is good to already insert the vectors we have determined @Azkle ?

**Comment by JWGmeligMeyling** [comment]
Solve the merge conflicts :smile:

File: src/main/java/me/moodcat/mood/Mood.java (hunk 5 above)
Comment (Azkle) [inline src/main/java/me/moodcat/mood/Mood.java]: I can try to determinate this vectors.
We hadn't written them down yet, but that should be feasible.
We just tried to determinate a vector for a song and that was a bit of "nattevingerwerk"/inaccurate.

**Pull request merged**